### PR TITLE
Add config parameters for traced models

### LINKED ISSUE #2566: [DOC] [ml-commons] add new model config parameters
**What do you want to do?**
 
- [x] Request a change to existing documentation
- [ ] Add new documentation
- [ ] Report a technical problem with the documentation
- [ ] Other 

**Tell us about your request.** Provide a summary of the request and all versions that are affected.
 
We add two new model parameters in 2.5 which not included in this doc https://opensearch.org/docs/latest/ml-commons-plugin/model-serving-framework/#the-model_config-object. 

- "pooling_mode": type "string",  If user trace model with Huggingface transformer, they may need to specify how to post-process model output by specifying "pooling_mode" and "normalize_result".  Supported pooling method: `mean`, `mean_sqrt_len`, `max`, `weightedmean`, `cls`. For example, this huggingface model https://huggingface.co/sentence-transformers/all-MiniLM-L6-v2 uses `mean` pooling mode.
- "normalize_result": type "boolean". If user trace model with Huggingface transformer, they may need to specify how to post-process model output by specifying "pooling_mode" and "normalize_result".  For example, this huggingface model https://huggingface.co/sentence-transformers/all-MiniLM-L6-v2 needs to normalize result, so need to set "normalize_result" as `true.

You can also read this doc for more details https://github.com/opensearch-project/ml-commons/blob/2.x/docs/model_serving_framework/text_embedding_model_examples.md,

**What other resources are available?** Provide links to related issues, POCs, steps for testing, etc.

### PR DESCRIPTION
**Pull request by Naarcha-AWS**
Fixes #2566 

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: _ml-commons-plugin/model-serving-framework.md [hunk 1]
```diff
@@ -68,6 +68,13 @@ Field | Data type | Description
 | `framework_type` | string  | The framework the model is using. Currently, we support `sentence_transformers` and `huggingface_transformers` frameworks. The `sentence_transformers` model outputs text embeddings directly, so ML Commons does not perform any post processing. For `huggingface_transformers`, ML Commons performs post processing by applying mean pooling to get text embeddings. See the example [`all-MiniLM-L6-v2` Huggingface model](https://huggingface.co/sentence-transformers/all-MiniLM-L6-v2) for more details. |
 | `all_config` _(Optional)_ | string | This field is used for reference purposes. You can specify all model configurations in this field. For example, if you are using a Huggingface model, you can minify the `config.json` file to one line and save its contents in the `all_config` field. Once the model is uploaded, you can use the get model API operation to get all model configurations stored in this field. |
 
+When using a pre-trained model, the following fields are required in the `model_config` object.
```
Comment (ylwu-amzn) [inline _ml-commons-plugin/model-serving-framework.md:71]: These two parameters are optional, read this doc https://github.com/opensearch-project/ml-commons/blob/2.x/docs/model_serving_framework/text_embedding_model_examples.md#11-trace-sentence-transformers-model

> Sentence transformer model already includes post-processing logic. So no need to specify pooling_mode/normalize_result when upload model.

> Without [sentence-transformers](https://pypi.org/project/sentence-transformers/) installed, you can trace this model AutoModel.from_pretrained('sentence-transformers/all-MiniLM-L6-v2'). But model traced this way doesn't include post-processing. So user have to specify post-process logic with pooling_mode and normalize_result.

**Review by kolchfa-aws — APPROVED** [review]
LGTM.

File: _ml-commons-plugin/model-serving-framework.md [hunk 2]
```diff
@@ -68,6 +68,13 @@ Field | Data type | Description
 | `framework_type` | string  | The framework the model is using. Currently, we support `sentence_transformers` and `huggingface_transformers` frameworks. The `sentence_transformers` model outputs text embeddings directly, so ML Commons does not perform any post processing. For `huggingface_transformers`, ML Commons performs post processing by applying mean pooling to get text embeddings. See the example [`all-MiniLM-L6-v2` Huggingface model](https://huggingface.co/sentence-transformers/all-MiniLM-L6-v2) for more details. |
 | `all_config` _(Optional)_ | string | This field is used for reference purposes. You can specify all model configurations in this field. For example, if you are using a Huggingface model, you can minify the `config.json` file to one line and save its contents in the `all_config` field. Once the model is uploaded, you can use the get model API operation to get all model configurations stored in this field. |
 
+You can further customize a sentence transformer pre-trained model's post-processing logic with the following optional fields in the `model_config` object.
```
Comment (cwillum) [inline _ml-commons-plugin/model-serving-framework.md:71]: Does "... pre-trained sentence transformer model's post-processing logic ..." preserve the meaning of this?

File: _ml-commons-plugin/model-serving-framework.md [hunk 3]
```diff
@@ -68,6 +68,13 @@ Field | Data type | Description
...
+
+| Field | Data type | Description |
+| :--- | :--- | :--- |
+| `pooling_mode` | string | The post-process model output, either `mean`, `mean_sqrt_len`, `max`, `weightedmean`, or `cls`.|
```
Comment (cwillum) [inline _ml-commons-plugin/model-serving-framework.md:75]: Can't tell you why. But we typically capitalize data types: String, Boolean.

**Review by cwillum — APPROVED** [review]
Just a couple of comments. Looks good.

File: _ml-commons-plugin/model-serving-framework.md (hunk 2 above)
Comment (Naarcha-AWS) [reply _ml-commons-plugin/model-serving-framework.md:71]: Yes. I'll adjust.

**Pull request merged**